common/player.go

```diff
@@ -51,6 +50,43 @@ type Player struct {
 	IsReloading                 bool
 	HasDefuseKit                bool
 	HasHelmet                   bool
+
+	// Note: this shouldn't be epxosed (it's just used internally), but i don't know how to do it ?
+	ThrownGrenades			map[int64]*Equipment
```

Comment:
Let's avoid having this on the `Player` struct as it's only for internal use,
Create `gameState.thrownGrenades` of type `map[*common.Player]map[EquipmentElement]*Equipment`.

Not sure if that solution is ideal but let's see where it takes us, maybe we'll have to adapt again if there's an issue.

Reply:
Sounds great !

So this way i could move `player.GetThrownGrenade(wep EquipmentElement)` & `player.DeleteThrownGrenadeByType(wep EquipmentElement)` functions to private functions like `gameEventHandler.getThrownGrenade(p *common.Player, wep EquipmentElement)` & `gameEventHandler.deleteThrownGrenadeByType(p *common.Player, wep EquipmentElement)` (which are only used in game_events)

Reply:
exactly 🙂 

Maybe we will need to move it into a seperate, nested struct since `gameState` is getting sort of bloated.
But just move it there for now and let's see how it looks.